WordPress/Classes/ViewRelated/SuggestionsListViewModel.swift

```diff
@@ -123,15 +123,42 @@ import CoreData
         }
         switch suggestions {
         case .users(let userSuggestions):
+            let sortedUserSuggestions = Self.sort(
+                userSuggestions: userSuggestions, by: searchQuery
+            )
             return Self.searchResultByMovingProminentSuggestionsToTop(
-                userSuggestions: userSuggestions,
+                userSuggestions: sortedUserSuggestions,
                 prominentSuggestionsIds: prominentSuggestionsIds
             )
         case .sites(let siteSuggestions):
             return .sites(siteSuggestions)
         }
     }
 
+    /// Sort user suggestions by prefix first, then alphabetically. The collection is sorted first by checking if displayName begins with the provided prefix. The remaining items are sorted alphabetically by displayName. The prefix comparison is both case-insensitive and diacritic-insensitive.
+    /// - Parameter userSuggestions: The user suggestions collection to be sorted.
+    /// - Parameter prefix: The prefix to be used when checking the usernames and displayNames.
+    /// - Returns: The sorted user suggestion collection.
+    private static func sort(userSuggestions: [UserSuggestion], by prefix: String) -> [UserSuggestion] {
+        guard !userSuggestions.isEmpty, !prefix.isEmpty else { return userSuggestions }
+        var prefixedUserSuggestions = [UserSuggestion]()
+        var otherUserSuggestions = [UserSuggestion]()
+
+        for suggestion in userSuggestions {
+            guard let username = suggestion.username else {
+                continue
+            }
+            if let displayName = suggestion.displayName,
+               (username.beginsWith(prefix: prefix) || displayName.beginsWith(prefix: prefix)) {
+                prefixedUserSuggestions.append(suggestion)
```

Comment:
This will skip users with a `nil` `displayName`… even if they have a `username` that begins with the prefix.

So `{"ID":102,"user_login":"carlosdiaz","display_name":null}` will not be found even if `prefix` is `carl` here, because `let displayName = suggestion.displayName` will be false and exit the condition already.

Is that intentional? Do we want to prioritize only the users who have a non-nil display name?
If not, this condition will need to be adjusted accordingly 🙂

Reply:
Nice catch. The idea was to have users with a `username` or a `displayName` beginning with the prefix first. I think your dictionary approach suggestion will cover that. I will add a new entry to test this scenario, also.